doc: fixes/enhancements to k8s helm for appsmith-ee (#708)
appsmithorg/appsmith-docs: website/docs/getting-started/setup/installation-guides/kubernetes/README.mdx

    ```bash
    helm repo update
    ```
3. Generate the `values.yaml` config file
    ```bash
    helm show values appsmith-ee/appsmith --version 2.0.0 > values.yaml
    ```
4. Edit the `values.yaml` file and set the following parameters

   ```yaml
    applicationConfig:
      APPSMITH_ENCRYPTION_PASSWORD: "<PASSWORD>"
      APPSMITH_ENCRYPTION_SALT: "<SALT_PASSWORD>"
      APPSMITH_LICENSE_KEY: "<LICENSE_KEY>"

>    ```

Reviewer: The salt and password need to be filled in with randomly generated passwords. Can we add this detail. Currently, it doesn't indicate _what_ should be filled in for there variables.

Reviewer 2: sure

<details id="highAvailability" open>
  <summary> <b>To configure High Availability (HA), be sure to configure the following</b> </summary>

:::note Important

>   Both PostgreSQL and MongoDB must be enabled or configured to point to an external instance when `autoscaling` is enabled

Reviewer: ```suggestion
  Both PostgreSQL and MongoDB must be enabled or configured to point to an external instance when `autoscaling` is enabled.
```

:::

  1. Install `metrics-server`, which provides vital metrics to the Horizontal Pod Autoscaler (HPA) to scale the pods.

    ```bash
    kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml
    ```

  2. Enable `autoscaling` in the `values.yaml`. The below shows the `autoscaling` configuration for the Appsmith Deployment:

   ```yaml
        autoscaling:
          # highlight-next-line
          enabled: true
          minReplicas: 2
          maxReplicas: 2
   ```

  3. PostgreSQL has to be enabled Or configure an external PostgreSQL in the `applicationConfig` section in `values.yaml`.

>       i. Configure the parameters to enable PostgreSQL as shown below:

Reviewer: Please only use `1.`, `2.`, `3.` etc. Docusaurus will correctly convert it to `i.`, `ii.` and `iii.`. I know I'm repeating this point, but it creates very broken markup, where, instead of using list tags like `ol` and `li`, this will render as paragraphs, which is clearly not what we want here.

```suggestion
      1. Configure the parameters to enable PostgreSQL as shown below:
```

      ```yaml
            # enable PostgreSQL
            postgresql:
            # highlight-next-line
                enabled: true
                auth:
                  username: root
      ```

      ii. Configure the parameters to point to an external PostgreSQL as shown below:
    
      ```yaml
        # configure an external PostgreSQL
        applicationConfig::
          APPSMITH_KEYCLOAK_DB_DRIVER: "postgresql"
          APPSMITH_KEYCLOAK_DB_USERNAME: "<POSTGRES_USER>"
          APPSMITH_KEYCLOAK_DB_PASSWORD: "<POSTGRES_PASSWORD>"
          APPSMITH_KEYCLOAK_DB_URL: "<POSTGRES_DB_URL>"

>       ```

Reviewer: Same as above, change `ii.` to `2.`, and the alignment of the code block is also off.

  export CONTAINER_PORT=$(kubectl get pod --namespace default $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
  echo "Visit http://127.0.0.1:8080 to use your application"
  kubectl --namespace default port-forward $POD_NAME 8080:$CONTAINER_PORT
  ```

ii. Run the command `kubectl get pods` to verify the list of pods. You may see output as shown below:

  ```bash
  NAME                            READY   STATUS    RESTARTS   AGE
  appsmith-875b6cddc-4mmj6        1/1     Running   0          90s
  appsmith-875b6cddc-7b2hw        1/1     Running   0          3m30s
  appsmith-875b6cddc-smzwz        1/1     Running   0          3m15s
  appsmith-mongodb-0              1/1     Running   0          3m30s
  appsmith-mongodb-1              1/1     Running   0          3m10s
  appsmith-mongodb-arbiter-0      1/1     Running   0          3m30s
  appsmith-postgresql-0           1/1     Running   0          3m30s
  appsmith-redis-master-0         1/1     Running   0          3m30s
  appsmith-redis-replicas-0       1/1     Running   0          3m30s

>   ```

Reviewer: Here also, both listing syntax, as well as alignment, are off. Just check out how this renders. The nested list hierarchy is completely lost.

![Screenshot 2023-01-03 at 4 48 35 PM](https://user-images.githubusercontent.com/120119/210347562-3462e7ac-3176-495c-bdd2-9cdbdb9df4f6.png)

  <summary>Follow the steps below to configure using command</summary>
  Run the below command to enable ingress controller:

  ```bash
  helm upgrade appsmith-ee/appsmith \
      --set ingress.enabled=true \
      --set ingress.className=nginx \
      --set service.type=ClusterIP \
  ```

   </details>
   <details id="usingValues">
  <summary>Follow the steps below to configure using `values.yaml`</summary>
  
  Open the `values.yaml` file and update the parameter values as below:

  ```yaml 
  ingress:
    ## @param ingress.enabled Enable ingress record generation for Ghost
    ##
    enabled: true
    ## @param ingress.annotations Additional custom annotations for the ingress record
    ## NOTE: If `ingress.certManager=true`, annotation `kubernetes.io/tls-acme: "true"` will automatically be added
    ##
    className: nginx
    hosts: []
      # - host: appsmith-domain.me
  service:
    ## @param service.type Kubernetes Service type
    ##
    type: ClusterIP
  ```
    
Once the parameter values are modified, run the below command:

  ```bash 
  helm upgrade --values values.yaml appsmith appsmith-ee/appsmith
  ```
    
  </details>

</TabItem>
</Tabs>
  

> * It takes a few minutes for the ingress to get an IP assigned. Once it has been assigned, you should see an IPV4 address or a FQDN in the ADDRESS column. Run the below command to verify the assigned IP address/FQDN:

Reviewer: ```suggestion
* It takes a few minutes for the ingress to get a public IP assigned. Once it has been assigned, you should see an IPv4 address or a domain name in the `ADDRESS` column. Run the below command to get this address:
```

2. Add the following to the data section.

  ```yaml
  # Add the data section if not already present

>   data: 

Reviewer: Can we move the comment `# Add the data section...` outside the code block and into the above point text please? People often copy-paste contents of code-block, which means they'll also copy this comment. But this comment won't hold value, when pasted in the resource file. It's just documentation text.

The following command uninstalls Appsmith and removes all Kubernetes resources associated with the chart:

> 1. List the releases by running below command:

Reviewer: Can we check the alignment of this list items too please?
